sanic/server/protocols/websocket_protocol.py

```diff
@@ -0,0 +1,165 @@
+from typing import TYPE_CHECKING, Optional, Sequence, Union
+
+from websockets.connection import CLOSED, CLOSING, OPEN
+from websockets.server import ServerConnection
+
+from sanic.exceptions import ServerError
+from sanic.log import error_logger
+from sanic.server import HttpProtocol
+
+from ..websockets.impl import WebsocketImplProtocol
+
+
+if TYPE_CHECKING:
+    from websockets import http11
+
+
+class WebSocketProtocol(HttpProtocol):
+
+    websocket: Union[None, WebsocketImplProtocol]
+    websocket_timeout: float
+    websocket_max_size = Union[None, int]
+    websocket_ping_interval = Union[None, float]
+    websocket_ping_timeout = Union[None, float]
+
+    def __init__(
+        self,
+        *args,
+        websocket_timeout: Optional[float] = 10.0,
+        websocket_max_size: Optional[int] = None,
+        websocket_max_queue: Optional[int] = None,  # max_queue is deprecated
+        websocket_read_limit: Optional[int] = None,  # read_limit is deprecated
+        websocket_write_limit: Optional[int] = None,  # write_limit deprecated
+        websocket_ping_interval: Optional[float] = 20.0,
+        websocket_ping_timeout: Optional[float] = 20.0,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.websocket = None
+        self.websocket_timeout = websocket_timeout
+        self.websocket_max_size = websocket_max_size
+        if websocket_max_queue is not None and int(websocket_max_queue) > 0:
+            error_logger.warning(
+                DeprecationWarning(
+                    "websocket_max_queue is no longer used. "
+                    "No websocket message queueing is implemented."
+                )
+            )
+        if websocket_read_limit is not None and int(websocket_read_limit) > 0:
+            error_logger.warning(
+                DeprecationWarning(
+                    "websocket_read_limit is no longer used. "
+                    "No websocket rate limiting is implemented."
+                )
+            )
+        if (
+            websocket_write_limit is not None
+            and int(websocket_write_limit) > 0
+        ):
+            error_logger.warning(
+                DeprecationWarning(
+                    "websocket_write_limit is no longer used. "
+                    "No websocket rate limiting is implemented."
+                )
+            )
```

Comment:
Can you add an internal comment here that acts as a reminder to remove these notices after the LTS? We can remove them in v22.3

Reply:
Yep. Can do.